marklogic-data-hub/src/main/java/com/marklogic/hub/job/JobManager.java

```diff
@@ -85,6 +101,153 @@ public JobDeleteResponse deleteJobs(String jobIds) {
         return this.jobDeleteRunner.deleteJobs(jobIds);
     }
 
+    /**
+     * Export Job documents and their associated Trace documents to a zip file.
+     *
+     * @param exportFilePath specifies where the zip file will be written
+     * @param jobIds a comma-separated list of jobIds; if null, all will be exported
+     */
+    public void exportJobs(Path exportFilePath, String jobIds) {
+        File zipFile = exportFilePath.toFile();
+        WriteToZipConsumer zipConsumer = new WriteToZipConsumer(zipFile);
+
+        String[] jobsArray = jobIds != null ? jobsArray = jobIds.split(",") : null;
+
+        QueryManager qm = jobClient.newQueryManager();
+
+        // Build a query that will match everything
+        StringQueryDefinition emptyQuery = qm.newStringDefinition();
+        emptyQuery.setCriteria("");
+
+        // Get the job(s) document(s)
+        StructuredQueryBuilder sqb = qm.newStructuredQueryBuilder();
+        DataMovementManager dmm = jobClient.newDataMovementManager();
+        QueryBatcher batcher = null;
+        StructuredQueryDefinition query = null;
+        if (jobsArray == null) {
+            batcher = dmm.newQueryBatcher(emptyQuery);
+        }
+        else {
+            batcher = dmm.newQueryBatcher(sqb.value(sqb.jsonProperty("jobId"), jobsArray));
+        }
+        batcher.onUrisReady(new ExportListener().onDocumentReady(zipConsumer));
+        JobTicket jobTicket = dmm.startJob(batcher);
+
+        batcher.awaitCompletion();
+        dmm.stopJob(batcher);
+        dmm.release();
+
+        JobReport report = dmm.getJobReport(jobTicket);
+        long jobCount = report.getSuccessEventsCount();
+
+        if (jobCount > 0) {
+            DatabaseClient client = getTraceDatabaseClient();
+
+            // Get the traces that go with the job(s)
+            dmm = client.newDataMovementManager();
+            if (jobsArray == null) {
+                batcher = dmm.newQueryBatcher(emptyQuery);
+            }
+            else {
+                batcher = dmm.newQueryBatcher(sqb.value(sqb.element(new QName("jobId")), jobsArray));
+            }
+            batcher.onUrisReady(new ExportListener().onDocumentReady(zipConsumer));
+            dmm.startJob(batcher);
+
+            batcher.awaitCompletion();
+            dmm.stopJob(batcher);
+            dmm.release();
+
+            zipConsumer.close();
+        }
+        else {
+            // there were no jobs, so don't produce an empty zip file
+            zipConsumer.close();
+            zipFile.delete();
+        }
+
+    }
+
+    /**
+     * We create this new client instead of using traceClient because the DataMovementManger relies on an
+     * internal-only endpoint, and tracing-rewriter.xml doesn't account for it. As such, we're using the
+     * Jobs app server, but pointing to the Traces database.
+     * @return
+     */
+    private DatabaseClient getTraceDatabaseClient() {
```

Comment:
Look at HubConfig.newTraceDbCient()

I put those new*Client() methods there so that we don't duplicate the creation code. Also, it handles SSL and stuff. This approach will likely fail for ssl.

Reply:
I think I started with that, but the Trace app server uses a non-standard rewriter that doesn't include some hidden URLs. DMSDK failed because of that. We talked about re-copying the rewriter stuff, but that doesn't feel like a good approach long term. 

Won't the `jobClient.getSecurityContext()` figure out the SSL situation?

Reply:
that rewriter is moving into the server install dir and will be maintained like the regular rest rewriter. So we can count on it being ok. Erik says there is a magical tool that checks to make sure all the various rewriter*.xml files are in sync when a server build is created.

Reply:
Okay, cool. I'll update the writer for now and switch to using newTraceDbClient